cmd/tink-worker/client/client_test.go

```diff
@@ -0,0 +1,222 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/packethost/pkg/grpc"
+	"github.com/packethost/pkg/log"
+	"github.com/tinkerbell/tink/protos/workflow"
+	ggrpc "google.golang.org/grpc"
+	"google.golang.org/grpc/test/bufconn"
+)
+
+// Path to a self-signed cert we use to test credential generation.
+const selfSignedCertFile string = "self-signed-cert.pem"
+
+type fakeServer struct {
+	data map[string]string
+	err  error
+}
```

Comment:
If it's helpful/useful, here's a little trick: add `workflow.WorkflowServiceServer` to `fakeServer` and you wont need to stub out all the interface methods. Only the ones you explicitly call. And I don't see any interface method calls, so you would be able to remove all the method stubs.
```Go
type fakeServer struct {
	workflow.WorkflowServiceServer
	data map[string]string
	err  error
}
```

Reply:
Thanks, this one really helped to cut down on the code.